scripts/benchmark_multithreaded.py

```diff
@@ -69,98 +130,109 @@ def get_core_counts(args):
+            "--output",
+            result_file,
+            "--runs",
+            str(number_of_runs),
+            "--scheduler=" + use_scheduler,
+            "--cores",
+            str(core_count),
+            "--clients",
+            str(args.clients),
         ]
         if args.verbose:
-            execution_command.append('--verbose')
+            execution_command.append("--verbose")
         if hyrise_args:
             execution_command += hyrise_args
         execution_command += get_formatted_queries(args)
-        verbose_print(args.verbose, 'Executing command: ' + subprocess.list2cmdline(execution_command) + '\n')
+        verbose_print(args.verbose, "Executing command: " + subprocess.list2cmdline(execution_command) + "\n")
         subprocess.run(execution_command)
 
+
 def benchmark(args, hyrise_args):
-    result_name = args.result_name if args.result_name else datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+    result_name = args.result_name if args.result_name else datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
     result_dir = os.path.join(args.result_dir, result_name)
-        
+
     if not os.path.exists(result_dir):
         os.makedirs(result_dir)
 
     core_counts = get_core_counts(args)
-    verbose_print(args.verbose, 'Executing benchmarks with the following numbers of cores: ' + str(core_counts))
+    verbose_print(args.verbose, "Executing benchmarks with the following numbers of cores: " + str(core_counts))
 
     # Make sure our machine has the number of cores we are trying to benchmark with
     for core_count in core_counts:
-        assert core_count <= MAX_CORE_COUNT, 'Can\'t use more cores than we have available'
+        assert core_count <= MAX_CORE_COUNT, "Can't use more cores than we have available"
 
     run_benchmarks(args, hyrise_args, core_counts, result_dir)
 
-    verbose_print(args.verbose, 'Benchmarks complete!')
-    verbose_print(args.verbose, 'Saved JSON results to: ' + result_dir)
+    verbose_print(args.verbose, "Benchmarks complete!")
+    verbose_print(args.verbose, "Saved JSON results to: " + result_dir)
 
     return result_dir
 
+
 def plot(args, result_dir):
-    verbose_print(args.verbose, 'Plotting results from: ' + result_dir)
+    verbose_print(args.verbose, "Plotting results from: " + result_dir)
 
     utilized_cores_per_numa_node = []
     max_cores = 0
 
     results = {}
     for _, _, files in os.walk(result_dir):
-        json_files = [f for f in files if f.split('.')[-1] == 'json']
-        num_cores_from_filename = lambda filename: int(filename.split('-')[0])
+        json_files = [f for f in files if f.split(".")[-1] == "json"]
         # Add the results in sorted order (low core count -> high core count) for plotting later
-        for file in sorted(json_files, key=num_cores_from_filename):
-            with open(os.path.join(result_dir, file), 'r') as json_file:
+        for file in sorted(json_files, key=lambda filename: int(filename.split("-")[0])):
```

Comment:
Was this something that the linter made you do?

Reply:
Yes, lambdas should not be assigned to variables. If we want to give it a name to make it more descriptive it should be a function.

Reply:
I'd do that or add a comment containing the filename magic. Right now, it is unclear what we are doing here.